ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,339 @@
+
+To use custom values in an Ingress rule define this annotations:
+
+`ingress.kubernetes.io/upstream-max-fails`: number of unsuccessful attempts to communicate with the server that should happen in the duration set by the fail_timeout parameter to consider the server unavailable
+
+`ingress.kubernetes.io/upstream-fail-timeout`: time in seconds during which the specified number of unsuccessful attempts to communicate with the server should happen to consider the server unavailable. Also the period of time the server will be considered unavailable.
+
+**Important:** 
+The upstreams are shared. i.e. Ingress rule using the same service will use the same upstream. 
+This means only one of the rules should define annotations to configure the upstream servers
+
+
+Please check the [custom upstream check](examples/custom-upstream-check/README.md) example
+
+
+### Authentication
+
+Is possible to add authentication adding additional annotations in the Ingress rule. The source of the authentication is a secret that contains usernames and passwords inside the the key `auth`
+
+The annotations are:
+
+```
+ingress.kubernetes.io/auth-type:[basic|digest]
+```
+
+Indicates the [HTTP Authentication Type: Basic or Digest Access Authentication](https://tools.ietf.org/html/rfc2617).
+
+```
+ingress.kubernetes.io/auth-secret:secretName
+```
+
+Name of the secret that contains the usernames and passwords with access to the `path/s` defined in the Ingress Rule.
+The secret must be created in the same namespace than the Ingress rule
+
+```
+ingress.kubernetes.io/auth-realm:"realm string"
+```
+
+Please check the [auth](examples/custom-upstream-check/README.md) example
+
+
+### Rewrite
+
+In some scenarios the exposed URL in the backend service differs from the specified path in the Ingress rule. Without a rewrite any request will return 404.
+To allow this is possible to use the annotation `ingress.kubernetes.io/rewrite-target` with the path expected in the service.
+If the application contains relative links is possible to add an additional annotation `ingress.kubernetes.io/add-base-url` that will append a `base` tag in the header of the returned HTML from the backend.
+
+
+Please check the [rewrite](examples/rewrite/README.md) example
+
+
+### Rate limiting
+
+The annotations `ingress.kubernetes.io/limit-connections` and `ingress.kubernetes.io/limit-rps` allows the creation of a limit in the connections that can be opened by a single client IP address. This can be use to mitigate [DDoS Attacks](https://www.nginx.com/blog/mitigating-ddos-attacks-with-nginx-and-nginx-plus)
+
+`ingress.kubernetes.io/limit-connections`: number of concurrent allowed connections from a single IP address
+
+`ingress.kubernetes.io/limit-rps`: number of allowed connections per second from a single IP address
+
+
+Is possible to specify both annotation in the same Ingress rule. In this case the order of the limit is `limit-connections` and then `limit-rps`
```

Comment:
If you specify both annotations in a single Ingress rule, limit-rps takes precedence